src/foundry/common/prosemirror/highlight-matches-plugin.d.mts

```diff
@@ -74,5 +75,5 @@ declare class ProseMirrorHighlightMatchesPlugin extends ProseMirrorPlugin {
    */
   constructor(schema: Schema, options: ProseMirrorMenu.Options);
 
-  static override build(schema: Schema, options?: Record<string, unknown> | undefined): Plugin;
+  static override build(schema: Schema, options?: EmptyObject | undefined): Plugin;
```

Comment:
So `| undefined` in an optional function parameter has no effect. The only difference is with object properties. So you don't need `| undefined` here or in other functions.